### ISSUE TITLE
Update watcher to handle reverted relays

### ISSUE THREAD

**Issue by TransmissionsDev** [ethereum-optimism/optimism#602]
[This PR](https://github.com/ethereum-optimism/optimism/pull/596) was just merged which adds an event that gets emitted when a relayed message reverts.

Would be nice to be able to pick up this event using the watcher to see information about how the message reverted, etc.

Should we add a new method for getting a reverted relay message or just detect the fail event in the existing `getL2TransactionReceipt` method?

**Comment by TransmissionsDev** [comment]
cc @smartcontracts

**Comment by platocrat** [comment]
As per #741, this issue was also experienced by Synthetix in their integration tests.

**Comment by platocrat** [comment]
@funfair-tech is also still blocked on this

cc: @snario

**Comment by snario** [comment]
#745 is related to this, though suggests an alternative approach

**Comment by TransmissionsDev** [comment]
> #745 is related to this, though suggests an alternative approach

Even if that is implemented the watcher still needs to be updated. Not mutually exclusive.

**Comment by smartcontracts** [comment]
This was silently fixed in https://github.com/ethereum-optimism/optimism/pull/937

Forgot about this issue. Please note that the watcher doesn't distinguish between failed and successful messages. For L1 => L2 messages you can distinguish based on the transaction receipt (status == 0 -> failed message). If you think it'd be worth distinguishing the two I'd recommend opening another issue with your preferred interface. Ty!!!

**Comment by TransmissionsDev** [comment]
> This was silently fixed in #937
> 
> Forgot about this issue. Please note that the watcher doesn't distinguish between failed and successful messages. For L1 => L2 messages you can distinguish based on the transaction receipt (status == 0 -> failed message). If you think it'd be worth distinguishing the two I'd recommend opening another issue with your preferred interface. Ty!!!

Great yeah this is how I thought it should be all along!
